### PR TITLE
Integer Conversion Issues from GCC

### PR DESCRIPTION
**Pull request by franksinankaya**
Debugger is being linked with -fPIE but dependent libraries are compiled with -fPIC. GCC is having trouble linking these.

@am11 @janvorli @jkotas

### CODE REVIEW

File: src/debug/createdump/CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,7 @@ include_directories(BEFORE ${VM_DIR})
 
 add_definitions(-DPAL_STDCPP_COMPAT=1)
 
-add_compile_options(-fPIE)
+add_compile_options(-fPIC)
```
Comment (janvorli) [inline src/debug/createdump/CMakeLists.txt]: The `createdump` is an executable, so  it should be compiled with `-fPIE`. I don't see where we link in code compiled with `-fPIC`. The `createdump_lib` is linked with '-fPIE' and the `mscordaccore` is a shared library.
Looking at the command line of compiling one of the sources above, I also don't see `-fPIC` being specified there, so it is not clear to me why does GCC complain.

Comment (franksinankaya) [reply src/debug/createdump/CMakeLists.txt]: I agree, I'll back out this change to figure out what exactly is happening.

**Comment by franksinankaya** [comment]
ping for merge

**Comment by bendono** [comment]
This looks like it belongs to the SOS plugin. My understanding is that this will move to the Diagnostics repository.
Should probably ping @mikem8361 .

**Comment by janvorli** [comment]
Yes, this change will need to be copied to the diagnostics repo. @mikem8361 when do we plan to remove the SOS stuff from the coreclr repo so that we don't need to make changes like this on two places?

**Comment by mikem8361** [comment]
We will probably remove sos from coreclr in the preview6 time frame. We are still waiting for the "new" SOS to shipping with the next public release of the Windows debugger (windbg) and there are still some OS/architecture holes in the diagnostics repo's build (and test) matrix (i.e. arm, arm64, FreeBSD, etc.).

I'll move these changes to the diagnostics repo.

/cc: @tommcdon

**Pull request merged**